src/util-exception-policy.c

```diff
@@ -69,36 +81,86 @@ void ExceptionPolicyApply(Packet *p, enum ExceptionPolicy policy, enum PacketDro
                 DecodeSetNoPacketInspectionFlag(p);
                 break;
         }
+    } else {
+        switch (policy) {
+            case EXCEPTION_POLICY_NOT_SET:
+                break;
+            case EXCEPTION_POLICY_BYPASS_FLOW:
+                PacketBypassCallback(p);
+                /* fall through */
+            case EXCEPTION_POLICY_PASS_FLOW:
+                SCLogDebug("EXCEPTION_POLICY_PASS_FLOW");
+                if (p->flow) {
+                    p->flow->flags |= FLOW_ACTION_PASS;
+                    FlowSetNoPacketInspectionFlag(p->flow); // TODO util func
+                }
+                /* fall through */
+            case EXCEPTION_POLICY_PASS_PACKET:
+                SCLogDebug("EXCEPTION_POLICY_PASS_PACKET");
+                DecodeSetNoPayloadInspectionFlag(p);
+                DecodeSetNoPacketInspectionFlag(p);
+                break;
+            default:
+                break;
+        }
```

Comment:
This looks ugly to me, due to the code repetition. But I couldn't quickly come up with a better alternative - other than maybe having two separate functions, but in the end, they'd still have pretty similar code...